client/src/pages/profile/index.tsx

```diff
@@ -22,53 +21,38 @@ import PreScreeningIviewCard from 'components/Profile/PreScreeningIviewCard';
 import { withGoogleMaps } from 'components/withGoogleMaps';
 import { NotificationChannel, NotificationsService } from 'modules/Notifications/services/notifications';
 import { ProfileInfo, ProfileMainCardData, UserService } from 'services/user';
-import { ActiveCourseProvider, SessionProvider } from 'modules/Course/contexts';
+import { ActiveCourseProvider, SessionContext, SessionProvider } from 'modules/Course/contexts';
 
-type Props = {
-  router: NextRouter;
-  session: Session;
+type ConnectionValue = {
+  value: string;
+  enabled: boolean;
+  lastLinkSentAt?: string;
 };
 
-type State = {
-  profile: ProfileInfo | null;
-  isProfileOwner: boolean;
-  isLoading: boolean;
-  isSaving: boolean;
-  connections: Partial<
-    Record<
-      NotificationChannel,
-      | {
-          value: string;
-          enabled: boolean;
-        }
-      | undefined
-    >
-  >;
-};
+type Connections = Partial<Record<NotificationChannel, ConnectionValue | undefined>>;
 
 export type ChangedPermissionsSettings = {
   permissionName: string;
   role: string;
 };
 
 const profileApi = new ProfileApi();
+const userService = new UserService();
+const notificationsService = new NotificationsService();
 
-export class ProfilePage extends React.Component<Props, State> {
-  state: State = {
-    profile: null,
-    isProfileOwner: false,
-    isLoading: true,
-    isSaving: false,
-    connections: {},
-  };
-
-  private userService = new UserService();
-  private notificationsService = new NotificationsService();
+const ProfilePage = () => {
+  const router = useRouter();
+  const session = useContext(SessionContext);
+  const [profile, setProfile] = useState<ProfileInfo | null>(null);
+  const [isProfileOwner, setIsProfileOwner] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isSaving, setIsSaving] = useState(false);
+  const [connections, setConnections] = useState<Connections>({});
 
-  private hadStudentCoreJSInterview = (stats: StudentStats[]) =>
+  const hadStudentCoreJSInterview = (stats: StudentStats[]) =>
```

Comment:
I'd move helper functions(which do not rely on state/props) outside of the component.